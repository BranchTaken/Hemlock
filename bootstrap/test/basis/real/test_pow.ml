open! Basis.Rudiments
open! Basis
open Real

let test () =
  let rec fn pairs = begin
    match pairs with
    | [] -> ()
    | (b, x) :: pairs' -> begin
        let xf = of_sint x in
        File.Fmt.stdout
        |> Fmt.fmt "** pow int_pow "
        |> fmt ~alt:true ~radix:Radix.Hex b
        |> Fmt.fmt " ~p:"
        |> Sint.pp x
        |> Fmt.fmt " -> "
        |> fmt ~alt:true ~radix:Radix.Hex (b ** xf)
        |> Fmt.fmt " "
        |> fmt ~alt:true ~radix:Radix.Hex (pow b ~p:xf)
        |> Fmt.fmt " "
        |> fmt ~alt:true ~radix:Radix.Hex (int_pow b ~p:x)
        |> Fmt.fmt "\n"
        |> ignore;
        fn pairs'
      end
  end in
  fn [
    (3., Sint.kv (-3L));
    (-1., Sint.kv 61L);
    (1., Sint.kv 61L);
    (2., Sint.kv (-1L));
    (2., Sint.kv 0L);
    (2., Sint.kv 1L);
    (2., Sint.kv 2L);
    (2., Sint.kv 61L);
    (10., Sint.kv 7L);
    ((ex 1.), Sint.kv (-1L));
    ((ex 1.), Sint.kv 0L);
    ((ex 1.), Sint.kv 1L);
    ((ex 1.), Sint.kv 2L);
  ]

let _ = test ()
