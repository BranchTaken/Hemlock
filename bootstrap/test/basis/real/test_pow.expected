** pow int_pow 0x3. ~p:-3i -> 0x0.097b_425e_d097_b4 0x0.097b_425e_d097_b4 0x0.097b_425e_d097_b4
** pow int_pow -0x1. ~p:61i -> -0x1. -0x1. -0x1.
** pow int_pow 0x1. ~p:61i -> 0x1. 0x1. 0x1.
** pow int_pow 0x2. ~p:-1i -> 0x0.8 0x0.8 0x0.8
** pow int_pow 0x2. ~p:0i -> 0x1. 0x1. 0x1.
** pow int_pow 0x2. ~p:1i -> 0x2. 0x2. 0x2.
** pow int_pow 0x2. ~p:2i -> 0x4. 0x4. 0x4.
** pow int_pow 0x2. ~p:61i -> 0x1p61 0x1p61 0x1p61
** pow int_pow 0xa. ~p:7i -> 0x98_9680. 0x98_9680. 0x98_9680.
** pow int_pow 0x2.b7e1_5162_8aed_2 ~p:-1i -> 0x0.5e2d_58d8_b3bc_e 0x0.5e2d_58d8_b3bc_e 0x0.5e2d_58d8_b3bc_e
** pow int_pow 0x2.b7e1_5162_8aed_2 ~p:0i -> 0x1. 0x1. 0x1.
** pow int_pow 0x2.b7e1_5162_8aed_2 ~p:1i -> 0x2.b7e1_5162_8aed_2 0x2.b7e1_5162_8aed_2 0x2.b7e1_5162_8aed_2
** pow int_pow 0x2.b7e1_5162_8aed_2 ~p:2i -> 0x7.6399_2e35_376b_4 0x7.6399_2e35_376b_4 0x7.6399_2e35_376b_4
